### ISSUE TITLE
Psalm should warn about phpdoc param annotations that don't correspond to real params.

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#448]
Expected: Warn about $notX not corresponding to a param.
Observed: No warning.

I forget if I've filed this already, didn't see any search results.

Another example is src/Psalm/Plugin.php

```php
<?php
/**
 * @param int $x
 * @param int $notX
 */
function test($x) : void {
    return;
}

class A3 {
    /**
     * @param int $x
     * @param int $notX
     */
    function test($x) : void {
        return;
    }
}
```

**Comment by muglug** [comment]
For reference, it was added here: https://github.com/vimeo/psalm/commit/b6eea4e812ad93eb60787dd7b680ade8bc46452b when fixing a related issue

And removed here: https://github.com/vimeo/psalm/commit/e29dd140e3dce9eae997368d6ce00e19b6fcdcf1#diff-135af77803f5d36d069031a46e3cf485
